streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -113,8 +116,8 @@ void openExisting(final ProcessorContext context) {
     }
 
     List<Segment> segments(final long timeFrom, final long timeTo) {
-        final long segFrom = segmentId(Math.max(0L, timeFrom));
-        final long segTo = segmentId(Math.min(currentSegmentId * segmentInterval, Math.max(0, timeTo)));
+        final long segFrom = Math.max(minSegmentId, segmentId(Math.max(0L, timeFrom)));
+        final long segTo = Math.min(currentSegmentId, segmentId(Math.min(currentSegmentId * segmentInterval, Math.max(0, timeTo))));
```

Comment:
nit: shall we rename `currentSegmentId` to sth. like `latestSegmentId` or `maxSegmentId`?